contrib/mesos/pkg/executor/executor.go

```diff
@@ -388,53 +380,21 @@ func (k *Executor) LaunchTask(driver bindings.ExecutorDriver, taskInfo *mesos.Ta
 	// (a) we ignore future launches for it
 	// (b) we have a record of it so that we can kill it if needed
 	// (c) we're leaving podName == "" for now, indicates we don't need to delete containers
+	var launchTimer *time.Timer
+	if k.launchGracePeriod > 0 {
+		launchTimer = time.NewTimer(k.launchGracePeriod)
+	} else {
+		launchTimer = &time.Timer{}
```

Comment:
Seems the logic in this class would be simpler if you just always created a timer with launchGracePeriod duration. The constructor automatically sets zero and negative durations to "now".